core/hostname/hostname_whitebox_test.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2013 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
```

Comment:
This file should be called `hostname_test.go`.

We've had that convention for some time, we don't want to introduce a new one.